automated/android/noninteractive-tradefed/tradefed.sh

```diff
@@ -82,31 +87,33 @@ if [ -e "${TEST_PATH}/testcases/vts/testcases/kernel/linux_kselftest/kselftest_c
     sed -i "/suspend/d" "${TEST_PATH}"/testcases/vts/testcases/kernel/linux_kselftest/kselftest_config.py
 fi
 
-if [ -n "${AP_SSID}" ] && [ -n "${AP_KEY}" ]; then
-    # try to connect to wifi with the feature provided by tradefed by default
-    # when AP_SSID and AP_KEY are specified for this tradefed test action explicitly
-    TEST_PARAMS="${TEST_PARAMS} --wifi-ssid ${AP_SSID} --wifi-psk ${AP_KEY}"
-else
-    # try to connect to wifi with the external AdbJoinWifi apk from
-    # https://github.com/steinwurf/adb-join-wifi
-    # if AP_SSID and AP_KEY are not specified for this tradefed test action
-    adb_join_wifi "${AP_SSID}" "${AP_KEY}"
-fi
+if [ "X${INTERNET_ACCESS}" = "Xtrue" ]; then
+    if [ -n "${AP_SSID}" ] && [ -n "${AP_KEY}" ]; then
+        # try to connect to wifi with the feature provided by tradefed by default
+        # when AP_SSID and AP_KEY are specified for this tradefed test action explicitly
+        TEST_PARAMS="${TEST_PARAMS} --wifi-ssid ${AP_SSID} --wifi-psk ${AP_KEY}"
+    else
+        # try to connect to wifi with the external AdbJoinWifi apk from
+        # https://github.com/steinwurf/adb-join-wifi
+        # if AP_SSID and AP_KEY are not specified for this tradefed test action
+        adb_join_wifi "${AP_SSID}" "${AP_KEY}"
+    fi
 
-# wait for a while till the wifi connecting operation finished
-sleep 60
+    # wait for a while till the wifi connecting operation finished
+    sleep 60
```

Comment:
white space

Reply:
hmm, It's white space here on my local side. Or you mean to change to use tab?